### DISCUSSION TITLE
Subscriber Cold Start?

### DISCUSSION THREAD

**Discussion by Atanusaha143** [ag2ai/faststream#1957] [Q&A]
I’ve been working with FastStream and everything has been running smoothly until I encountered an unusual issue. If my FastStream app runs for an extended period without processing any messages for 2–3 hours, the subscriber fails to process the first message it receives after this idle period. However, subsequent messages are processed successfully. 

Has anyone else faced a similar issue or knows what might be causing this? Any insights would be greatly appreciated!

```python
class Subscriber:
    @nats_broker.subscriber(
        subject="some_subject_name",
        stream="some_stream_name",
        durable="some_consumer_name",
        pull_sub=PullSub(batch=True, batch_size=5),
        no_ack=True, 
        retry=True,
    )
    async def handle_subscriber(self, msg: NatsMessage, logger: Logger):
        try:
            logger.info("Started processing requests for 'some_subject_name'.")

            decoded_messages = Subscriber._decode_messages(msg=msg)
            logger.debug(f"Decoded messages: {decoded_messages}")

            message_dtos = Subscriber._map_message_dtos(decoded_messages=decoded_messages)
            logger.debug(f"Mapped message DTOs: {message_dtos}")

            await msg.ack()
            logger.info("Successfully processed requests.")
        except Exception as e:
            logger.error("An exception occurred during processing requests.", exc_info=e)
            await msg.nack()

    @staticmethod
    def _decode_messages(msg: NatsMessage):
        decoded_messages: list = []
        for message in msg.body:
            decoded_message = json.loads(message.decode("utf-8"))
            decoded_messages.append(decoded_message)
        return decoded_messages

    @staticmethod
    def _map_message_dtos(decoded_messages):
        pass
```

**Comment by Atanusaha143** [comment]
@Lancetnik Hi, apologies for mentioning you without prior notice, but it’s a bit urgent, and was hoping you could help.

**Comment by Lancetnik** [comment]
Hi! Could you show any logs representing the problem?

**Comment by Atanusaha143** [comment]
@Lancetnik The issue was solved when I have added a timeout=0.5. But lately I found another interesting problem and that is on Google Cloud Run the FastStream App takes ~10 mins to start. Here are the [logs](https://drive.google.com/file/d/1qIdpoOygDqoSf-uYViB4IJcBPACX0EUw/view)
